stages/org.osbuild.dnf5.sbom.spdx

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/python3
+import json
+import os
+import sys
+import tempfile
+
+import libdnf5
+
+import osbuild
+from osbuild.util.sbom.dnf5 import dnf_pkgset_to_sbom_pkgset
+from osbuild.util.sbom.spdx import sbom_pkgset_to_spdx2_doc
+
+
+def get_installed_packages(tree):
+    with tempfile.TemporaryDirectory() as tempdir:
+        base = libdnf5.base.Base()
+
+        conf = base.get_config()
+        conf.installroot = tree
+        conf.config_file_path = f"{tree}{conf.config_file_path}"
+        conf.pluginpath = f"{tree}{conf.pluginpath}"
+        conf.pluginconfpath = f"{tree}{conf.pluginconfpath}"
+        conf.persistdir = f"{tree}{conf.persistdir}"
+        conf.transaction_history_dir = f"{tree}{conf.transaction_history_dir}"
+        conf.system_cachedir = f"{tempdir}{conf.system_cachedir}"
+        conf.varsdir = [f"{tree}{d}" for d in conf.varsdir]
+        conf.reposdir = [f"{tree}{d}" for d in conf.reposdir]
+        conf.cachedir = f"{tempdir}{conf.cachedir}"
```

Comment:
Still not a fan of using string formatting to join paths.

Reply:
Sure, but the reason is that all those `conf.*` paths are absolute, so joining them with `os.path.join()` would require additional code, which I'm not a fan of. But I could overcome this feeling :shrug:

Reply:
I noticed that as well, but it's consistent with the dnf4 version of the PR so I think it's fine for now and maybe something to clean up eventually :)

Reply:
I've created https://github.com/osbuild/osbuild/issues/1964